addon/components/course-materials.js

```diff
@@ -1,126 +1,156 @@
+  onSlmSort() {},
+  typesWithUrl: Object.freeze(['file', 'link']),
+
+  isClmLoading: reads('courseLearningMaterialObjects.isPending'),
+  isSlmLoading: reads('sessionLearningMaterialObjects.isPending'),
+
+  courseLearningMaterialObjects: computed(function() {
+    const promise = this.fetchCourseLearningMaterials();
+    return PromiseArray.create({ promise });
   }),
-  sessionLearningMaterialObjects: computed('sessionLearningMaterials.[]', function(){
-    return new RSVPPromise(resolve =>{
-      this.get('sessionLearningMaterials').then(slms => {
-        map(slms.toArray(), sessionLearningMaterial => {
-          return new RSVPPromise(resolve =>{
-            sessionLearningMaterial.get('learningMaterial').then(learningMaterial => {
-              sessionLearningMaterial.get('session').then(session => {
-                session.get('firstOfferingDate').then(firstOfferingDate => {
-                  let obj = EmberObject.create({
-                    title: learningMaterial.get('title'),
-                    description: learningMaterial.get('description'),
-                    author: learningMaterial.get('originalAuthor'),
-                    type: learningMaterial.get('type'),
-                    url: learningMaterial.get('url'),
-                    citation: learningMaterial.get('citation'),
-                    sessionTitle: session.get('title'),
-                    firstOfferingDate,
-                  });
-                  resolve(obj);
-                });
-              });
-            });
-          });
-        }).then(lmObjects => {
-          resolve(lmObjects);
-        });
-      });
-    });
+
+  sessionLearningMaterialObjects: computed(function() {
+    const promise = this.fetchSessionLearningMaterials();
+    return PromiseArray.create({ promise });
   }),
-  filteredSessionLearningMaterialObjects: computed('filter', 'sessionLearningMaterialObjects.[]', function(){
-    return new RSVPPromise(resolve => {
-      this.get('sessionLearningMaterialObjects').then(objs => {
-        const filter = this.get('filter');
-        if (isEmpty(filter)) {
-          resolve(objs);
-        } else {
-          let exp = new RegExp(filter, 'gi');
-          let filteredObjs = objs.filter(obj => {
-            return (isPresent(obj.get('title')) && obj.get('title').match(exp)) ||
-                   (isPresent(obj.get('description')) && obj.get('description').match(exp)) ||
-                   (isPresent(obj.get('author')) && obj.get('author').match(exp)) ||
-                   (isPresent(obj.get('type')) && obj.get('type').match(exp)) ||
-                   (isPresent(obj.get('citation')) && obj.get('citation').match(exp)) ||
-                   (isPresent(obj.get('sessionTitle')) && obj.get('sessionTitle').match(exp));
-          });
-
-          resolve(filteredObjs);
-        }
-      });
-    });
+
+  filteredCourseLearningMaterialObjects: computed(
+    'courseLearningMaterialObjects', 'clmQuery', function() {
```

Comment:
`courseLearningMaterialObjects` returns a PromiseArray. I commented obove that it should return a promise. In either case this method should be async or depend on `courseLearningMaterialObjects.[]`.